src/PowerShellEditorServices.Hosting/EditorServicesRunner.cs

```diff
@@ -0,0 +1,276 @@
+            (Stream inStream, Stream outStream) = await _config.DebugServiceTransport.ConnectStreamsAsync().ConfigureAwait(false);
+
+            _logger.Log(PsesLogLevel.Diagnostic, "Creating debug adapter");
+            return _serverFactory.CreateDebugServerWithLanguageServer(inStream, outStream, languageServer);
+        }
+
+        private async Task<PsesDebugServer> RecreateDebugServer(PsesDebugServer debugServer)
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Recreating debug adapter transport");
+            (Stream inStream, Stream outStream) = await _config.DebugServiceTransport.ConnectStreamsAsync().ConfigureAwait(false);
+
+            _logger.Log(PsesLogLevel.Diagnostic, "Recreating debug adapter");
+            return _serverFactory.RecreateDebugServer(inStream, outStream, debugServer);
+        }
+
+        private async Task<PsesDebugServer> CreateDebugServerForTempSession(HostStartupInfo hostDetails)
+        {
+            (Stream inStream, Stream outStream) = await _config.DebugServiceTransport.ConnectStreamsAsync().ConfigureAwait(false);
+
+            return _serverFactory.CreateDebugServerForTempSession(inStream, outStream, hostDetails);
+        }
+
+        private HostStartupInfo CreateHostStartupInfo()
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Creating startup info object");
+
+            (string allUsersProfilePath, string currentUserProfilePath) = GetProfilePaths(_config.ProfilePaths?.AllUsersProfilePath, _config.ProfilePaths?.CurrentUserProfilePath);
+
+            return new HostStartupInfo(
+                _config.HostInfo.Name,
+                _config.HostInfo.ProfileId,
+                _config.HostInfo.Version,
+                _config.PSHost,
+                allUsersProfilePath,
+                currentUserProfilePath,
+                _config.FeatureFlags,
+                _config.AdditionalModules,
+                _config.LogPath,
+                (int)_config.LogLevel,
+                consoleReplEnabled: _config.ConsoleRepl != ConsoleReplKind.None,
+                usesLegacyReadLine: _config.ConsoleRepl == ConsoleReplKind.LegacyReadLine);
+        }
+
+        private (string allUsersPath, string currentUserPath) GetProfilePaths(string allUsersPath, string currentUserPath)
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Configuring profile paths");
+
+            if (allUsersPath == null || currentUserPath == null)
+            {
+                using (var pwsh = SMA.PowerShell.Create())
+                {
+                    _logger.Log(PsesLogLevel.Diagnostic, "Querying PowerShell for profile paths");
+                    Collection<string> profiles = pwsh.AddScript("$profile.AllUsersAllHosts,$profile.CurrentUserAllHosts")
+                        .Invoke<string>();
+
+                    if (allUsersPath == null)
+                    {
+                        allUsersPath = profiles[0];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'allUsersPath'.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)